jupyterlab/labapp.py

```diff
@@ -897,13 +897,15 @@ def initialize(self, argv=None):
                 import jupyter_collaboration  # noqa
             except ImportError:
                 self.log.critical(
-                    """
+                    """\
+Exiting due to inability to start collaboration:
+
 To enable real-time collaboration, you must install the extension `jupyter_collaboration`.
 You can install it using pip for example:
 
   python -m pip install jupyter_collaboration
 
-This flag is now deprecated and will be removed in JupyterLab v5.
+To run without collaboration, disable collaboration (for example, remove the `--collaborative` flag from the commandline; this flag is now deprecated and will be removed in JupyterLab v5).  To see more ways to adjust the collaborative behavior, see https://discourse.jupyter.org/t/how-to-disable-jupyter-collaboration-extension/21329 and https://discourse.jupyter.org/t/how-to-disable-jupyter-collaboration-extension/21329/9
```

Comment:
Thanks.  I rebased the PR branch onto `main` and pushed it again because some UI tasks failed (? hope the rebase fixes).

I put the documentation link you suggested first, but included the second discussion link because it has some cross-cutting information that is very useful in practice.

I can remove and leave just the documentation link in; for ease of review, I include screenshots of and links to the two links in this comment:

Documentation at https://jupyterlab-realtime-collaboration.readthedocs.io/en/latest/configuration.html :

![image](https://github.com/jupyterlab/jupyterlab/assets/777015/36a3bd34-f583-4132-9a07-edef761b3106)


Discussion at  https://discourse.jupyter.org/t/how-to-disable-jupyter-collaboration-extension/21329/9 :

![image](https://github.com/jupyterlab/jupyterlab/assets/777015/9976be3b-4976-4d2a-85c2-05d6ed8b75fe)

Reply:
> I can remove and leave just the documentation link in; for ease of review, I include screenshots of and links to the two links in this comment:

Please do remove the link to the forum, I also believe we should not need to link to it. If in future there is a specific bit of information which is on the forum but not in the docs, the  docs need to be updated, because the docs are versioned and permanent, while the forum answers only reflect what is working as of the time they were written.

Reply:
I'm hoping that, as a long-time jupyter lab user myself, the statement that "this forum post was useful because it brought together configuration information that is spread across the documentation"  would help clarify why it's useful.

> If [...] there is [...] information [...] not in the docs, the docs need to be updated

The issue is not that the information is not **present** in the documentation, but that it's not **in the same place** in the documentation.  The documentation structure puts the information the user needs to fix this problem -- the information about how to affect/remove some configuration that they may have had already set -- in separate documentation sections, probably for good reasons.  No documentation structure can meet all the requirements of every issue, because issues can touch on different aspects of the system.  In this case, the issue revolves around configuration, and the configuration can exist in multiple places but the documentation documents these places in separate sections.

Reply:
Please see my other comment where I suggest a solution which "fixing" the documentation to include more examples and links to common Jupyter configuration docs.